### DISCUSSION TITLE
importing statement

### DISCUSSION THREAD

**Discussion by SquilliamX** [Cyfrin/foundry-full-course-cu#847] [Q&A] [answered]
Hi everyone, I am excited to announce that I have finished the course!! I am now working on my own personal project of flashloans. 
 I am excited to use announce that I will be using balancer as the flashloan provider and I want to use foundry as my development framework. In the balancer docs, it says to 
 
"import "@balancer-labs/v2-vault/contracts/interfaces/IVault.sol";
import "@balancer-labs/v2-vault/contracts/interfaces/IFlashLoanRecipient.sol";"

When i imported these statements I recieved this error code -

"Source "@balancer-labs/v2-interfaces/contracts/vault/IVault.sol" not found: File not found. Searched the following locations: "".solidity(6275)"

How can I fix this with foundry? I believe i can solve this easily with npm install but we didn't use npm install in the foundry course. I would assume that i can download the code into my lib(just like the foundry course) without npm. How can I do this? I can't find their github repo of that directory, so i am a bit lost. Can anyone help me?

This is the balancer docs- " https://docs.balancer.fi/guides/arbitrageurs/flash-loans.html "

**Comment by theirrationalone** [comment]
One thing you can do is, If those interfaces aren't available then You can copy those interfaces from official explorers i.e., etherscan, etc..
Still you'll need to set remappings with `@balanceer-labs/v2-vault/contracts/` to the folder where you'll place your copied interfaces/contracts/libraries files. That will help your protocol/contract to recognize required contracts on live chain.

**Comment by SquilliamX** [comment, marked as answer]
I am proud to say i figured it out!!! Instead of doing NPM install as the docs say, foundry is new so many people don't have the their docs in foundry. But simply enough, to set up the api, just go on their github and find what directory you need and then "forge install" the website endings into your project and set up the remappings. Thank you patrick collins for teaching me how to use foundry <3. foundry is the best framework for smart contract development!!

this is what it looked like for me and for anyone wanting to use foundry as the framework for balancer flashloans-

```
$ forge install balancer/balancer-v2-monorepo --no-commit
Installing balancer-v2-monorepo in "/home/smartchain/code/myprojects/first-flashloan/lib/balancer-v2-monorepo" (url: Some("https://github.com/balancer/balancer-v2-monorepo"), tag: None)
    Installed balancer-v2-monorepo
```

then i updated my remappings to 

```
remappings = [
    "@balancer-labs/v2-interfaces=lib/balancer-v2-monorepo/pkg/interfaces",
]
```

and now my import statement worked!!
